src/libraries/System.Private.CoreLib/src/System/IO/FileStream.Unix.cs

```diff
@@ -373,21 +373,7 @@ private Task FlushAsyncInternal(CancellationToken cancellationToken)
                 return Task.FromException(e);
             }
 
-            // We then separately flush to disk asynchronously.  This is only
-            // necessary if we support writing; otherwise, we're done.
-            if (CanWrite)
-            {
-                return Task.Factory.StartNew(
-                    static state => ((FileStream)state!).FlushOSBuffer(),
-                    this,
-                    cancellationToken,
-                    TaskCreationOptions.DenyChildAttach,
-                    TaskScheduler.Default);
-            }
-            else
-            {
-                return Task.CompletedTask;
-            }
```

Comment:
https://github.com/dotnet/coreclr/pull/24902 removes `FlushOSBuffer` call only from `FileStream.Windows.cs`; it looks to me that `FileStream.Unix.cs` should also get this change.

Reply:
At first glance, removing this code makes sense to me since the whole method is identical to the one from Windows in the linked PR. But I would like @stephentoub to confirm if he has any additional context on this proposed change. There might have been a reason why the linked PR only removed the code for Windows, but I don't see it described in the description.

Reply:
I don't know why I didn't delete this as part of that PR, but I should have.  So, yes, we should delete it here.

That said, FlushAsyncInternal is now identical between FileStream.Windows.cs and FileStream.Unix.cs.  The FlushAsyncInternal from the .Windows.cs file should be moved into FileStream.cs and then this whole function in FileStream.Unix.cs deleted.